### PR TITLE
feat: workspaces

### PR DESCRIPTION
**Pull request by bodinsamuel**
- Move everything into workspaces to avoid dependencies conflict happening in new eslint version
- Upgrade dependencies
- Changed a bit the extends ability 
- Add few missing rules

Releases script has been tested just until actual release, it might be broken.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 language: node_js
-node_js: stable
+node_js:
+  - '14.15.4'
```
Comment (Haroenv) [inline .travis.yml:3]: let's change this to circle in a next PR?

File: packages/eslint-config-algolia/flowtype.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // eslint-disable-next-line import/no-commonjs
 module.exports = {
-  extends: ['plugin:flowtype/recommended', './base.js', 'prettier/flowtype'],
+  extends: ['plugin:flowtype/recommended'],
   plugins: ['flowtype'],
+  parser: 'babel-eslint',
```
Comment (Haroenv) [inline packages/eslint-config-algolia/flowtype.js:5]: let's remove this config in a next PR/manor

File: packages/eslint-config-algolia/typescript.js [hunk 3]
```diff
@@ -9,9 +9,5 @@ module.exports = {
       jsx: true,
     },
   },
-  extends: [
-    './base.js',
-    './rules/typescript.js',
-    'prettier/@typescript-eslint',
```
Comment (Haroenv) [inline packages/eslint-config-algolia/typescript.js:15]: the prettier line is gone now for typescript, have you tested if it's not needed anymore?

File: packages/eslint-config-algolia/vue.js [hunk 4]
```diff
@@ -19,15 +19,8 @@ module.exports = {
       jsx: true,
     },
   },
-  extends: [
-    'plugin:vue/strongly-recommended',
-    'eslint:recommended',
-    'plugin:import/errors',
-    './rules/base.js',
-    'prettier',
```
Comment (Haroenv) [inline packages/eslint-config-algolia/vue.js:27]: prettier seems to be removed now

File: scripts/release.sh [hunk 5]
```diff
@@ -50,17 +57,20 @@ read -e newVersion
 printf "\n\nRelease: generate TOCS"
 doctoc README.md --maxlevel 3
 
+cd packages/eslint-config-algolia
 npm version "$newVersion" --no-git-tag-version
```
Comment (Haroenv) [inline scripts/release.sh:61]: let's also update the root package.json's version after here?

File: package.json [hunk 6]
```diff
@@ -1,29 +1,16 @@
...
-    "lint": "eslint . --ignore-pattern sample-project",
-    "lint:fix": "npm run lint -- --fix"
-  },
-  "keywords": [
-    "eslint",
-    "eslintconfig",
-    "es6",
-    "algolia"
-  ],
+  "version": "1.0.0",
```
Comment (Haroenv) [inline package.json:2]:
```suggestion
  "version": "16.0.0",
```

File: packages/eslint-config-algolia/package.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    ],
+    "semanticCommits": true,
+    "rebaseStalePrs": true,
+    "automerge": true,
+    "major": {
+      "automerge": false
+    },
+    "automergeType": "branch-push"
+  },
+  "homepage": "https://github.com/algolia/eslint-config-algolia",
```
Comment (Haroenv) [inline packages/eslint-config-algolia/package.json:35]:
```suggestion
  "homepage": "https://github.com/algolia/eslint-config-algolia",
```

File: packages/test/package.json [hunk 8]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "test",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "private": true,
+  "keywords": [],
```
Comment (Haroenv) [inline packages/test/package.json:7]:
```suggestion
  "name": "test",
  "version": "1.0.0",
  "main": "index.js",
  "private": true,
```

**Review by Haroenv — COMMENTED** [review]
cool changes!

File: packages/eslint-config-algolia/typescript.js (hunk 3 above)
Comment (bodinsamuel) [reply packages/eslint-config-algolia/typescript.js:15]: it's not unneeded, it's not working anymore with the new prettier plugin.

File: packages/eslint-config-algolia/flowtype.js (hunk 2 above)
Comment (bodinsamuel) [reply packages/eslint-config-algolia/flowtype.js:5]: you mean the entire file flowtype.js ?

Comment (Haroenv) [reply packages/eslint-config-algolia/flowtype.js:5]: yes, I don't think we use flow anywhere anymore

File: packages/eslint-config-algolia/vue.js (hunk 4 above)
Comment (bodinsamuel) [reply packages/eslint-config-algolia/vue.js:27]: one of the goal of this PR is to be able to combine extends and avoir redeclaration, as the doc changes suggest prettier is now in base `extends: ['algolia', 'algolia/vue']`

File: .travis.yml (hunk 1 above)
Comment (bodinsamuel) [reply .travis.yml:3]: actually I revert to a LTS, stable currently mean 15.0 which is not long time supported

Comment (Haroenv) [reply .travis.yml:3]: sorry, I meant changing from travis to circleci

**Review by Haroenv — APPROVED** [review]
let's try this in a major version!

File: .travis.yml (hunk 1 above)
Comment (bodinsamuel) [reply .travis.yml:3]: ah yes indeed !

**Pull request merged**